hasher-matcher-actioner/hmalib/lambdas/api/api_auth.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import os
+import json
+import time
+import jwt
+import requests
+import base64
+from jwt.algorithms import RSAAlgorithm
+
+from hmalib.common.logging import get_logger
+
+ACCESS_TOKEN = os.environ["ACCESS_TOKEN"]
```

Comment:
Blocking: I think we should store this in an AWS secret rather than an environment variable

Pros:
- Users can control access easier with IAM permissions
- Its encrypted
- Users don't have to re-depoly terraform every time they want to change the token and the token can be changed by users who might not have access to the terraform deployment. If we eventually enable users to create/generate API ACCESS_TOKENs from the UI, this lambda would have to be redeployed on every change

Reply:
These changes are incremental, I like your suggestion and have thought of it as well but is adds another layer complexity and would have greatly increased the size of the PR. More over it does not actually change the way one would integrate just how the token value is set.    

As is this addresses the need for a permanent token right now and keeps the door open for this: all we'd need to do is change the impl of `validate_access_token` to query aws secret. 

Aside: an apply is not a full re-depoly, and is very easy to update without require any changes or use of the aws console.